src/software/api/runner.html

```diff
@@ -384,488 +396,583 @@
+                            hide();
+                            loginFormHasBeenShown = true;
+                            loginSpinner.style.display = 'none';
+                            shouldAutoLoginDemo = false;
+                        }, function (error) {
+                            console.log('authenticateCallback: error');
+                            loginError.innerHTML = error;
+                            loginError.style.display = "block";
+                            passwordInput.select();
+                            passwordInput.focus();
+                            loginSpinner.style.display = 'none';
+                            shouldAutoLoginDemo = false;
+                        });
+                    }
+
+                    loginButton.removeEventListener("click", prevLoginClickHandler, false);
+                    loginClickHandler = debounce(loginClickHandler);
+                    prevLoginClickHandler = loginClickHandler;
+                    loginButton.addEventListener("click", loginClickHandler, false);
+                };
+
+                const preventDefault = e => e.preventDefault();
+                let prevLoginClickHandler = null;
+
+                authenticateManager = debounce(authenticateManager);
 
                 processors.add("js", EditorManager("js", "javascript", resizeObserver, urlState));
                 processors.add("css", EditorManager("css", "css", resizeObserver, urlState));
                 processors.add("html", EditorManager("html", "html", resizeObserver, urlState));
 
                 layout.resize();
 
+                // Process a search string if there is one. The only use for the search string is to automatically login a user
+                // to a local geotabdemo db. This feature is used by geo-cli so that users can easily open up an api runner
+                // for their local geo-cli-managed dbs. It is for internal development uses only and will only work with local dbs.
+                let searchString = window.location.search;
+                if (searchString) {
+                    try {
+                        if (searchString?.startsWith('?')) {
+                            // The search string will look like this:
+                            // ?server=localhost:10001&database=geotabdemo&username=dawsonmyers@geotab.com&password=passwordpassword
+                            searchString = searchString.replaceAll('?', '').replaceAll('\n', '').replaceAll('\\n', '');
+                            // Decode the query string if it is base64 encoded.
+                            const decodedSearchString = searchString.includes('=') ? searchString : atob(searchString);
+                            console.log(`decodedSearchString: ${decodedSearchString}`);
+
+                            const urlParams = new URLSearchParams(decodedSearchString);
+
+                            const requiredParams = ['database', 'username', 'server', 'password'];
```

Comment:
I'm a little wary about sending password as part of the url params. Is it possible we could use session id instead?